pkg/algorithm/cstorpoolselect/v1alpha1/select.go

```diff
@@ -102,22 +211,122 @@ type preferAntiAffinityLabel struct {
 
 // name returns the name of this policy
 func (p preferAntiAffinityLabel) name() policyName {
-	return preferAntiAffinityLabelPolicyName
+	return preferAntiAffinityLabelPolicy
 }
 
 // filter piggybacks on antiAffinityLabel policy
 // with the difference being; this logic returns all
 // the provided pools if there are no pools that
 // satisfy antiAffinity rule
-func (p preferAntiAffinityLabel) filter(poolUIDs []string) ([]string, error) {
-	plist, err := p.antiAffinityLabel.filter(poolUIDs)
+func (p preferAntiAffinityLabel) filter(pools *csp.CSPList) (*csp.CSPList, error) {
+	plist, err := p.antiAffinityLabel.filter(pools)
 	if err != nil {
-		return nil, err
+		return plist, err
 	}
-	if len(plist) > 0 {
+	if len(plist.Items) > 0 {
 		return plist, nil
 	}
-	return poolUIDs, nil
+	return pools, nil
+}
+
+type executionMode string
+
+const (
+	// multiExection enables execution of
+	// more than one policy during a selection
+	multiExecution executionMode = "multi-mode"
+
+	// singleExecution enables execution of
+	// only one policy during a seclection
+	singleExection executionMode = "single-mode"
+)
+
+type policyList struct {
+	items map[priority][]policy
+}
+
+func (pl *policyList) getAll() []policy {
+	if len(pl.items) == 0 {
+		return nil
+	}
+	var all []policy
+	for _, policies := range pl.items {
+		all = append(all, policies...)
+	}
+	return all
+}
+
+func (pl *policyList) add(p policy) {
+	pl.items[p.priority()] = append(pl.items[p.priority()], p)
+}
+
+func (pl *policyList) sortByPriority() []policy {
+	var sorted []policy
+	if len(pl.items) == 0 {
+		return sorted
+	}
+	for i := highPriority; i >= 0; i-- {
+		if len(pl.items[i]) == 0 {
+			continue
+		}
+		sorted = append(sorted, pl.items[i]...)
+	}
+	return sorted
+}
+
+type policyListPredicate func(*policyList) bool
+
+func hasPolicy(name string) policyListPredicate {
```

Comment:
change the signature to make use of policyName instead of string.
```go
func hasPolicy(p policyName) policyListPredicate {
  // ...
}
```

Reply:
done